### DISCUSSION TITLE
Links between given list of devices

### DISCUSSION THREAD

**Discussion by imranomar** [netbox-community/netbox#10868] [Q&A] [answered]
Hi All. The question I would like to pose is how can I get a list ( as an api ) of links between devices. The netbox-ui-plugin does a good job in creating a topology diagram of how devices are linked by just pass a list of deivces but I need the links information to be passed through as api so we can know how the devices are linked.

Thanking you
Imran

![image](https://user-images.githubusercontent.com/2282183/200744910-1c32bdf3-5340-4a3d-a238-fa362193f127.png)

**Comment by candlerb** [comment]
You need to search for "Cables". Assuming it's the REST API you're talking about, then the endpoint is `/dcim/cables/`. You can filter on `device` or `device_id` to limit to cables into a particular device or list of devices - repeat the query parameter for multiple devices. You can also filter by cables which terminate in a given rack, location or site. See source ref [here](https://github.com/netbox-community/netbox/blob/v3.3.7/netbox/dcim/filtersets.py#L1594-L1638).

Note that the cable model changed substantially between Netbox 3.2 and 3.3, so if you're not yet on 3.3, it would be better to upgrade to this first before spending too much time on it.

Note also that the Cables model only gives individual cable segments, so if device A is connected to device B via patch panels P and Q, you'll find three separate cables A-P, P-Q, Q-B.  You may find it useful to try the `/dcim/connected-device/` endpoint, or the various `.../trace/` endpoints (e.g. `/dcim/interfaces/{id}/trace/`)

**Reply by imranomar** [reply]
Thanks Brian

**Comment by imranomar** [comment, marked as answer]
I managed to resolve the issue by creating another django app and creating a class that extends ApiVIew for the view. I imported the get_topology from nextbox_ui_plugin.views and passed the device_ids using the queryset, filterset and it returns devices information with the topology links.
